pkg/ha/doc.go

```diff
@@ -0,0 +1,146 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ha
+
+// Data need to be persisted for the HA scheduler.
+// - configuration:
+//   - the upstream MySQL config (content of `MysqlConfig`):
+//     - PUT when adding an upstream (`operate-worker create`) by DM-master.
+//       - verify the validation before PUT it into etcd.
+//     - GET when scheduling the source to a DM-worker instance by DM-worker.
+//     - DELETE when removing an upstream (`operate-worker stop`) by DM-master.
+//       - DELETE with `the expectant stage of the relay` in one txn.
+//       - DELETE with `the bound relationship between the DM-worker instance and the upstream MySQL source` in one txn.
+//     - TODO: UPDATE support with `the expectant stage of the relay`.
+//   - the data migration task config (content of `TaskConfig`):
+//     - PUT when starting a task (`start-task`) by DM-master.
+//       - verify the validation before PUT it into etcd.
+//       - PUT with `the expectant stage of the subtask` in one txn.
+//     - GET when starting a subtask by DM-worker.
+//     - DELETE when stopping a task (`stop-task`) by DM-master.
+//       - DELETE with `the expectant stage of the subtask` in one txn.
+//     - TODO: UPDATE support with `the expectant stage of the subtask`.
+//
+// - node information (name, address, etc.):
+//   - the DM-worker instance:
+//     - PUT when adding a DM-worker instance by DM-master.
+//     - GET only when restoring the in-memory information after the leader of DM-master changed by the new leader.
+//     - DELETE when removing a DM-worker instance by DM-master.
+//     - TODO: UPDATE support later.
+//
+// - the health status (or keep-alive) of component instances:
+//   - the DM-worker instance:
+//     - PUT (keep-alive) by DM-worker (when the node is healthy).
+//     - GET (through WATCH) by DM-master to know is another schedule needed.
+//     - DELETE (when the lease is timeout) by etcd (when the node is un-healthy).
+//     - no need to UPDATE it manually.
+//
+// - the running stage:
+//   - NOTE: persist the current stage of the relay and subtask if needed later.
+//   - the bound relationship between the DM-worker instance and the upstream MySQL source (including relevant relay and subtasks):
+//     - PUT when scheduling the source to a DM-worker instance by DM-master.
+//       - PUT with `the expectant stage of the relay` in one txn.
+//     - GET (through WATCH) by DM-worker to know relevant relay/subtasks have to do.
+//     - DELETE when removing an upstream by DM-master.
+//       - DELETE with `the upstream MySQL config` in one txn.
+//       - DELETE with `the expectant stage of the relay` in one txn.
+//     - UPDATE when scheduling the source to another DM-worker instance by DM-master.
```

Comment:
How to `UPDATE`?

Reply:
Besides, what's the form of the key of the running stage? `/sourceID/workerAddr` or `/workerAddr/sourceID`?

Reply:
PUT the new value to overwrite the old one.
May be `/sourceID` (key) -> `worker-name` (value), I'll add that later.